### DISCUSSION TITLE
Creating package that clones git repository and runs script to modify another application

### DISCUSSION THREAD

**Discussion by Infinidoge** [chocolatey/choco#2194] [Q&A]
Hello all, I'm new to creating chocolatey packages and am looking to create a few for software I used to install manually.
In this case, I am looking to create a package for [powercord](https://powercord.dev/), which injects into [discord](discord.com).

The [package for Discord Canary](https://chocolatey.org/packages/discord-canary) already exists on the community repo, so I don't believe I need to do anything there.

My question then is, based on the [powercord installation instructions](https://powercord.dev/installation), how would I go about managing the git repo, checking for updates, and installing them?

My current thought process is to clone the [powercord repository](https://github.com/powercord-org/powercord) into a folder then run the injection process as the instructions state, either cloning to a temporary folder to be removed later, or into a more permanent place while the installation exists. Though the latter option is preferable, I'm not sure where the folder would be, since it doesn't need to be in the path.

Uninstallation should be similarly possible, just entering the cloned repository and running the uninstallation command.

However, updating I am not as sure. Should the package be fixed to a certain commit, then update that referenced commit with an auto-updating package? Depending on the licensing of powercord, could the repo just be included in the package and updated in that same manner?

Any advice would be appreciated!

**Comment by sparkles-laurel** [comment]
Well, I think you cannot distribute applications that are not accepted to be used. Powercord seems to be an unofficial Discord client, and according to the [Discord Terms of Service](https://dis.gd/tos), you cannot write, use or distribute 3rd party (modded) Discord clients.
